src/main/java/com/k3project/demo/controller/UserController.java

```diff
@@ -1,46 +1,81 @@
 package com.k3project.demo.controller;
 
+import com.k3project.demo.repository.UserRepository;
+import com.k3project.demo.service.dto.UserDTO;
 import com.k3project.demo.entity.User;
 import com.k3project.demo.service.UserService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.List;
-import java.util.Optional;
-import java.util.UUID;
+import java.util.*;
+
+
 
-/* ako mohol byt prvy controller ked tu uz sa vola instancia servici
-co je optional
- */
 @RestController
-@RequestMapping(path ="api_01/users")
+//define that all methods will returned Body in serialized Json format
+@RequestMapping(path = "/api/v1", produces = MediaType.APPLICATION_JSON_VALUE)
 public class UserController {
-
+    private final UserRepository userRepository;
     private final UserService userService;
 
-    public UserController(UserService userService) {
+    public UserController(UserRepository userRepository, UserService userService) {
+        this.userRepository = userRepository;
         this.userService = userService;
     }
-    @GetMapping(value="/allUsers")
-    public List<User> findAllUsers(){
+
+    @GetMapping(value = "/allUsers")
```

Comment:
For every method: 
 1. Use ResponseEntity as return value
 2. IF on the request is required obj UserDTO, use @Validation
 3. Use userDTO instead of user!